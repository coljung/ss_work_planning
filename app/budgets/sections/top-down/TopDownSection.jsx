import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Tabs } from 'antd';
import SectionContainer from '../SectionContainer';
import commonCellValueRenderer from './CommonCellRenderer';
import execCellValueRenderer from './ExecCellRenderer';

// Sub Component
const TabPane = Tabs.TabPane;

export const TAB_EXEC_RECAP = 'exec';
export const TAB_TOTAL = 'total';
export const TAB_WOMEN = 'women';
export const TAB_MEN = 'men';

export default class TopDownSection extends Component {
    constructor(props) {
        super(props);

        const { activeKey } = this.props;

        this.state = {
            activeTab: activeKey || TAB_EXEC_RECAP,
            [TAB_EXEC_RECAP]: false,
            [TAB_TOTAL]: false,
            [TAB_WOMEN]: false,
            [TAB_MEN]: false,
        };

        this.handleTabChange = this.handleTabChange.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.tab !== this.props.tab) {
            const currentKey = this.state.activeTab;

            this.setState({
                [currentKey]: false,
                activeTab: nextProps.tab,
                [nextProps.tab]: true,
            });

            // this.props.changeTab(nextProps.tab);
        }
    }

    handleTabChange(newTabKey) {
        // set true to load tabbed component
        const { activeKey } = this.props;

        this.setState({
            [activeKey]: false,
            activeTab: newTabKey,
            [newTabKey]: true,
        });
        this.props.changeTab(newTabKey);
    }

    render() {
        const { budget, version, data } = this.props;
        const { activeTab } = this.state;
        return (
            <Tabs activeKey={activeTab} onChange={this.handleTabChange} animated={false}>
                <TabPane tab="Exec Recap" key={TAB_EXEC_RECAP}>
                    {(activeTab === TAB_EXEC_RECAP || this.state[TAB_EXEC_RECAP]) &&
                        <SectionContainer
                            budget={budget}
                            version={version}
                            data={data}
                            cellRenderer={execCellValueRenderer}
                            key={TAB_EXEC_RECAP}
                            view={TAB_EXEC_RECAP}
                        />
                    }
                </TabPane>
                <TabPane tab="Total" key={TAB_TOTAL}>
                    {(activeTab === TAB_TOTAL) &&
                        <SectionContainer
                            budget={budget}
                            version={version}
                            data={data}
                            cellRenderer={commonCellValueRenderer}
                            key={TAB_TOTAL}
                            view={TAB_TOTAL}
                        />
                    }
                </TabPane>
                <TabPane tab="Women" key={TAB_WOMEN}>
                    {(activeTab === TAB_WOMEN) &&
                        <SectionContainer
                            budget={budget}
                            version={version}
                            data={data}
                            cellRenderer={commonCellValueRenderer}
                            key={TAB_WOMEN}
                            view={TAB_WOMEN}
                        />
                    }
                </TabPane>
                <TabPane tab="Men" key={TAB_MEN}>
                    {(activeTab === TAB_MEN) &&
                        <SectionContainer
                            budget={budget}
                            version={version}
                            data={data}
                            cellRenderer={commonCellValueRenderer}
                            key={TAB_MEN}
                            view={TAB_MEN}
                        />
                    }
                </TabPane>
            </Tabs>
        );
    }
}

TopDownSection.propTypes = {
    activeKey: PropTypes.string.isRequired,
    budget: PropTypes.string.isRequired,
    data: PropTypes.array.isRequired,
    tab: PropTypes.string.isRequired,
    version: PropTypes.string.isRequired,
    changeTab: PropTypes.func.isRequired,
};
