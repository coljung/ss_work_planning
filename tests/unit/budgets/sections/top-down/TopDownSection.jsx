import React from 'react';
import Enzyme, { mount } from 'enzyme';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TopDownSection from '../../../../../app/budgets/sections/top-down/TopDownSection';

describe('TopDownSection', () => {
    it('should render base layout', () => {
        const output = shallow(
            <TopDownSection title="fsdfsdf">Hello Jest!</TopDownSection>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
