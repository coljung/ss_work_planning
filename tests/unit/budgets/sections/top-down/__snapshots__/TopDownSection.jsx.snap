// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopDownSection should render base layout 1`] = `
<Tabs
  activeKey="exec"
  animated={false}
  hideAdd={false}
  onChange={[Function]}
  prefixCls="ant-tabs"
>
  <TabPane
    key="exec"
    placeholder={null}
    tab="Exec Recap"
  >
    <Connect(withRouter(SectionContainer))
      cellRenderer={[Function]}
      key="exec"
      view="exec"
    />
  </TabPane>
  <TabPane
    key="total"
    placeholder={null}
    tab="Total"
  />
  <TabPane
    key="women"
    placeholder={null}
    tab="Women"
  />
  <TabPane
    key="men"
    placeholder={null}
    tab="Men"
  />
</Tabs>
`;
